basics/uri.adoc

```diff
@@ -27,12 +28,70 @@ The following is taken from https://datatracker.ietf.org/doc/html/rfc3986#append
  reserved = gen-delims / sub-delims
 ----
 
-== uProtocol URI
+<<uuri-specification>> diagram below provides an overview of the specifications that we will elaborate in this document. 
 
 .URI Specification
+[#uuri-specification]
 image::uri.drawio.svg[uProtocol URI Specification]
 
-=== Long Form
+
+== Data Model
+
+`UUri` type is used to represent the URI. It is composed of `UAuthority`, `UEntity`, and `UResource` types.
+
+ * Type name *MUST* be `UUri`
+ * *MUST* have-a <<UAuthority>>, <<UEntity>>, and <<UResource>> types 
+ * *MUST* have methods to build `UUri` from either a string or any combination of <<UAuthority>>, <<UEntity>>, and <<UResource>> types based on the URI specifications
+ * *MUST* have a method to build the an RPC URI passing `<<UAuthority>> and <<UEntity>> where <<UResource>> is set to `"rpc.response"`.
+ * *MUST* have a method to convert to and from string
+ * *MUST* support URI with and without scheme defined (in the string)
+
+ 
+NOTE: the term *_Fully Qualified URI_* refers to a long or short form URI that contains all the items (UAuthority, UEntity, and UResource)
+
+
+=== UAuthority
+`UAuthority` type is responsible for storing the uDevice and uDomain portion of the uProtocol URI.
+
+ * Type name *MUST* be `UAuthority`
+ * *MUST* contain `domain` and `device` member variables of type `std::string` with accessor methods of the same name to fetch each
+ * *MUST* be able to construct a `UAuthority` from a string
+ * *MUST* support detection of local vs remote URIs from `UAuthority`
+
+if the authority is an IP address:
+
+ * IP address *MUST* be stored in `device` member variable 
```

Comment:
Is it important to know that the device has an IP address, as apposed to some other sting defining a device?
If so, I would suggest a static factory method called fromIP.
UAuthority.fromIpAddress(ip address)

This makes it explicit. In addition, we can now add a hint to the UAuthority that can help resolvers, resolve the UAuthority from some short value in the _Fully Qualified URI_

Reply:
What would the uE do with the IP address?  The SDK needs to know it.  Can't we hide that behind the helper functions?  I can see this will need to be different on the MCU side of things where IP addresses might be hardcoded, but that is different.

Reply:
Removing this requirement, IP address is stored in InetAddress not to confuse with device and domain names